frontend/components/Corpora.js

```diff
@@ -0,0 +1,148 @@
+                title: newCorpusData.title,
+                description: newCorpusData.description
+            })
+        };
+        fetch("api/add_corpus", requestOptions)
+            .then(response => response.json())
+            .then(data => {
+                setCorporaData(corporaData => [...corporaData, data]);
+                setNewCorpusData({
+                    "title": "",
+                    "description": ""
+                });
+                setAddingCorpus(false);
+            });
+    };
+
+    const addCorpusModal = () => {
+        return (
+            <>
+                <button className="btn btn-primary mb-3"
+                    onClick={handleShowModal}>Add Corpus</button>
+                <Modal show={showModal} onHide={handleCloseModal}>
+                    <Modal.Header closeButton>Add Corpus</Modal.Header>
+                    <form onSubmit={handleSubmit}>
+                        <Modal.Body>
+                            <div className="row mb-3">
+                                <label htmlFor="title"
+                                    className="col-2 col-form-label">Title</label>
+                                <div className="col">
+                                    <input type="text" className="form-control"
+                                        id="title" value={newCorpusData.title}
+                                        onChange={handleTitleInputChange} required/>
+                                </div>
+                            </div>
+                            <div className="row mb-3">
+                                <label htmlFor="description"
+                                    className="col form-label">Description</label>
+                            </div>
+                            <div className="row">
+                                <div className="col">
+                                    <textarea row="4" className="form-control"
+                                        id="description" value={newCorpusData.description}
+                                        onChange={handleDescriptionInputChange}/>
+                                </div>
+                            </div>
+                        </Modal.Body>
+                        <Modal.Footer>
+                            <button className="btn btn-secondary"
+                                onClick={handleCloseModal}>Close</button>
+                            <button className="btn btn-primary" type="submit">Add</button>
+                        </Modal.Footer>
+                    </form>
+                </Modal>
+            </>
+        );
+    };
+
+    const corporaList = () => {
+        return (
+            <>
+                {corporaData.map((corpus, i) => (
+                    <div className="col-6 mb-3" key={i}>
+                        <div className="card">
+                            <div className="card-body">
+                                <h6>{corpus.title}</h6>
```

Comment:
This `<h6>` is likely too deep a header level for the page. It looks like the only other header on the page is a `<h1>`, so these semantically should be `<h2>`s. If you want to style them to look less impactful, I'd recommend going through CSS to do so. That way, we preserve both the visual style (through the CSS) and the semantics and accessibility (through the HTML).